lib/html_diff/differ.rb

```diff
@@ -0,0 +1,78 @@
+require 'nokogiri'
+require 'diffy'
+
+module HtmlDiff
+  class Differ
+    REPLACEMENTS = {
+      'https://www-origin.staging.publishing.service.gov.uk' => 'https://www.gov.uk',
+      'https://www-origin.publishing.service.gov.uk' => 'https://www.gov.uk',
+      'https://assets-origin.staging.publishing.service.gov.uk' => 'https://assets.digital.cabinet-office.gov.uk',
+      /https:\/\/assets\.digital\.cabinet-office\.gov\.uk\/specialist-frontend\/application-[0-9a-f]{32}\.js/ => 'https://assets.digital.cabinet-office.gov.uk/specialist-frontend/application-7463fa64f198b6568dc121dae41d44b1.js',
+    }
+
+    attr_reader :differing_pages
```

Comment:
because it's the only instance variable that needs to be accessed from outside this class.